ledger/store/src/helpers/rocksdb/internal/nested_map.rs

```diff
@@ -198,8 +198,11 @@ impl<
 
         // Ensure that the atomic batch is empty.
         assert!(self.atomic_batch.lock().is_empty());
-        // Ensure that the database atomic batch is empty.
-        assert!(self.database.atomic_batch.lock().is_empty());
+        // Ensure that the database atomic batch is empty; skip this check if the atomic
+        // override is enabled, as there may be pending storage operations.
+        if !self.database.atomic_override.load(Ordering::SeqCst) {
```

Comment:
We should introduce an abstraction for this call everywhere, so that it is used properly:
```
self.database.atomic_override.load(Ordering::SeqCst)
```

Reply:
https://github.com/AleoHQ/snarkVM/pull/2296/commits/14a113609d221751f2081afc8465eeca944d6d31